host-controller/src/main/java/org/jboss/as/host/controller/operations/HostShutdownHandler.java

```diff
@@ -128,6 +131,19 @@ public void execute(OperationContext context, ModelNode operation) throws Operat
             } catch (Exception e) {
                 throw HostControllerLogger.ROOT_LOGGER.noServerInstallationPrepared(productName);
             }
+
+            // check the presence of a client marker in our server installation and if so, returns its value so the client
+            // can detect whether he was launched from the same installation dir.
+            final Path cliMarker = environment.getHomeDir().toPath().resolve("bin").resolve("cli-marker");
```

Comment:
We should test that the Files.exist

Reply:
I think we can do it without checking the existence since we can build Path objects for unknown files since Path doesn't access to the filesystem, do you agree?